### PR TITLE
[16.0][FIX] hr_timesheet_sheet_attendance: Fix timezone handling when linking attendances

### PR DESCRIPTION
**Pull request by joel-oates**
**Summary**

  - Fix timezone handling in timesheet sheet `create` method when linking attendances
  - Previously, UTC attendance timestamps were compared directly against local date boundaries, causing attendances to be incorrectly linked or missed when the employee's timezone differs from UTC
  - The fix converts the timesheet date range to UTC using the employee's timezone before searching for matching attendances

  **Changes**

  - **models/hr_timesheet_sheet.py**: Convert date_start/date_end to UTC using employee timezone before searching for attendances
  - **tests/test_hr_timesheet_sheet_create.py**: Add comprehensive tests covering:
    - Basic UTC range linking
    - Timezone boundary conditions (next day, previous day)
    - Summer/winter time handling
    - Exclusion cases
    - Already-assigned attendance handling
    - Unchecked-out attendance handling

**Test plan**
  - [x] All existing tests pass
  - [x] New timezone boundary tests pass

### CODE REVIEW

**Review by alexey-pelykh — APPROVED** [review]
Good fix! Timezone handling when linking attendances to timesheets is a classic pitfall. Converting to employee timezone before comparing dates is the right approach. The test coverage is thorough too — boundary cases for winter/summer time, exclusion tests, already-assigned check. LGTM

File: hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py [hunk 1]
```diff
@@ -90,17 +92,33 @@ def check_employee_attendance_state(self):
     @api.model
     def create(self, vals):
         res = super(HrTimesheetSheet, self).create(vals)
+
+        # Get employee timezone
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py]: Please avoid comments when the code is self-explicative.

File: hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py [hunk 2]
```diff
@@ -90,17 +92,33 @@ def check_employee_attendance_state(self):
     @api.model
     def create(self, vals):
         res = super(HrTimesheetSheet, self).create(vals)
+
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py]: avoid empty lines inside methods

File: hr_timesheet_sheet_attendance/tests/test_hr_timesheet_sheet_create.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
+import datetime
+
+from .hr_timesheet_sheet_test_cases import HrTimesheetTestCases
+
+
+class TestHrTimesheetSheetCreate(HrTimesheetTestCases):
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_attendance/tests/test_hr_timesheet_sheet_create.py:6]: Can you consolidate tests in 1 or 2 cases at most? I think the bug is clear, it is just the create method not linking timesheets when the user is in a UTC+X time and the hours before the next day are less than X.

If we can just make the new tests cover only what it has been fixed it'd be easier to review. Thanks.

**Review by AaronHForgeFlow — COMMENTED** [review]
The fix works but some comments on the code

File: hr_timesheet_sheet_attendance/tests/test_hr_timesheet_sheet_create.py (hunk 3 above)
Comment (joel-oates) [reply hr_timesheet_sheet_attendance/tests/test_hr_timesheet_sheet_create.py:6]: Sure thing! Removed all tests for the existing functionallity, just the core test remains

Comment (AaronHForgeFlow) [reply hr_timesheet_sheet_attendance/tests/test_hr_timesheet_sheet_create.py:6]: Thank you! If you remove the empty lines inside python method I think this is ok to go

Comment (AaronHForgeFlow) [reply hr_timesheet_sheet_attendance/tests/test_hr_timesheet_sheet_create.py:6]: and please leave only one commit with [FIX] message, it can be this for example:

[FIX] hr_timesheet_sheet_attendance: Fix timezone handling when linking attendances

**Comment by AaronHForgeFlow** [comment]
Can you check why git says there are conflicts? Perhaps the branch needs a rebase

image

File: hr_timesheet_sheet_attendance/tests/test_hr_timesheet_sheet_create.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    def test_create_timezone_boundary_next_day(self):
+        """Test attendance at UTC date boundary that's next day in Brussels"""
+        # Attendance at Jan 14, 23:30 UTC = Jan 15, 00:30 Brussels (UTC+1)
+        # So it's Jan 15 in employee's timezone
+        attendance = self._create_attendance(
+            employee=self.employee,
+            checkIn=datetime.datetime(2019, 1, 14, 23, 30, 0),
+            checkOut=datetime.datetime(2019, 1, 15, 2, 0, 0),
+        )
+
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_attendance/tests/test_hr_timesheet_sheet_create.py]: can you remove this blank line?

File: hr_timesheet_sheet_attendance/tests/test_hr_timesheet_sheet_create.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        # Timesheet for Jan 15-20
+        timesheet = self.env["hr_timesheet.sheet"].create(
+            {
+                "employee_id": self.employee.id,
+                "date_start": datetime.date(2019, 1, 15),
+                "date_end": datetime.date(2019, 1, 20),
+            }
+        )
+
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_attendance/tests/test_hr_timesheet_sheet_create.py]: same

**Review by AaronHForgeFlow — APPROVED** [review]
Code + functional

**Review by AaronHForgeFlow — APPROVED** [review]
👍

**Comment by Saran440** [comment]
@joel-oates Could you check tests please?
